diff --git a/common/config b/common/config
index 936ac22..0872664 100644
--- a/common/config
+++ b/common/config
@@ -122,8 +122,11 @@ export MKFS_PROG="$(type -P mkfs)"
 export MOUNT_PROG="$(type -P mount)"
 [ "$MOUNT_PROG" = "" ] && _fatal "mount not found"
 
-export UMOUNT_PROG="$(type -P umount)"
-[ "$UMOUNT_PROG" = "" ] && _fatal "umount not found"
+if [ -z "$UMOUNT_PROG" ]; then
+    UMOUNT_PROG="$(type -P umount)"
+    [ "$UMOUNT_PROG" = "" ] && _fatal "umount not found"
+fi
+export UMOUNT_PROG
 
 export FSSTRESS_PROG="./ltp/fsstress"
 [ ! -x $FSSTRESS_PROG ] && _fatal "fsstress not found or executable"
diff --git a/common/rc b/common/rc
index 5c4429e..21f0718 100644
--- a/common/rc
+++ b/common/rc
@@ -4545,7 +4545,7 @@ init_rc()
 
 	# Sanity check that TEST partition is not mounted at another mount point
 	# or as another fs type
-	_check_mounted_on TEST_DEV $TEST_DEV TEST_DIR $TEST_DIR $FSTYP || exit 1
+	_check_mounted_on TEST_DEV $TEST_DEV TEST_DIR $TEST_DIR $FSTYP$FUSE_SUBTYP || exit 1
 	if [ -n "$SCRATCH_DEV" ]; then
 		# Sanity check that SCRATCH partition is not mounted at another
 		# mount point, because it is about to be unmounted and formatted.
